src/net/ossl.cc

```diff
@@ -85,6 +87,34 @@ class ossl_error : public std::runtime_error {
     }
 };
 
+static cert_info::bytes extract_x509_serial(X509* cert) {
+    constexpr size_t serial_max = 128;
+    const ASN1_INTEGER *serial_no = X509_get_serialNumber(cert);
+    const size_t serial_size = std::min(serial_max, (size_t)serial_no->length);
+    cert_info::bytes serial(cert_info::bytes::initialized_later{}, serial_size);
+    std::memcpy(serial.begin(), reinterpret_cast<cert_info::bytes::value_type*>(serial_no->data), serial_size);
+    serial.resize(serial_size);
```

Comment:
question: Is this resize necessary?  Since above you create it with the size `serial_size`

Reply:
Ya, and what I put here was a partial transcription of something Ben had in a POC version. Short answer: yes, this could use some love.

Reply:
Since the commonalities between both methods are pretty small i just kept this method. I don't believe the resize is necessary, so i removed it